pykeepass/pykeepass.py

```diff
@@ -132,27 +133,34 @@ def save(self, filename=None, transformed_key=None):
             transformed_key (:obj:`bytes`, optional): precomputed transformed
                 key.
         """
-        output = None
+
         if not filename:
             filename = self.filename
 
         if hasattr(filename, "write"):
-            output = KDBX.build_stream(
+            KDBX.build_stream(
                 self.kdbx,
                 filename,
                 password=self.password,
                 keyfile=self.keyfile,
                 transformed_key=transformed_key
             )
         else:
-            output = KDBX.build_file(
-                self.kdbx,
-                filename,
-                password=self.password,
-                keyfile=self.keyfile,
-                transformed_key=transformed_key
-            )
-        return output
+            # save to temporary file to prevent database clobbering
+            # see issues 223, 101
+            # FIXME python2 - use pathlib.Path.withsuffix
+            filename_tmp = filename + '.tmp'
```

Comment:
Wouldn't it make more sense to use the [`tempfile` module](https://docs.python.org/3/library/tempfile.html) here?

Reply:
Fixed